app.rb

```diff
@@ -47,7 +47,7 @@ def openid_configuration
 
   def token(code)
     json HTTParty.post(
-      openid_configuration[:token_endpoint],
+      ENV['IDP_TOKEN_URL'] || openid_configuration[:token_endpoint],
```

Comment:
At this point I think it would just be easier to back out the HTTParty change to embed basic auth credentials in the URL and just go back to providing basic auth credentials manually. The whole point of the autodiscovery is that we can change URLs in the IDP and clients don't need code or config changes to continue working